lib/Command/RecreateMasterKey.php

```diff
@@ -0,0 +1,320 @@
+						$this->encryptUsersFiles($user, $progress);
+						$progress->advance();
+					}
+					$userNo++;
+				}
+				$offset += $limit;
+			} while(count($users) >= $limit);
+		}
+	}
+
+	public function encryptUsersFiles($uid, ProgressBar $progress) {
+
+		$this->setupUserFS($uid);
+		$directories = [];
+		$directories[] =  '/' . $uid . '/files';
+
+		while($root = array_pop($directories)) {
+			$content = $this->rootView->getDirectoryContent($root);
+			foreach ($content as $file) {
+				$path = $root . '/' . $file['name'];
+				if ($this->rootView->is_dir($path)) {
+					$directories[] = $path;
+					continue;
+				} else {
+					if($this->encryptFile($path) === false) {
+						$progress->setMessage("encrypt files for user $uid: $path (already encrypted)");
+						$progress->advance();
+					}
+				}
+			}
+		}
+	}
+
+	public function encryptFile($path) {
+		$source = $path;
+		$target = $path . '.encrypted.' . time();
+
+		try {
+			$this->ISession->set('encryptAllCmd', true);
+			$this->rootView->copy($source, $target);
+			$this->rootView->rename($target, $source);
+			$this->ISession->remove('encryptAllCmd');
+		} catch (DecryptionFailedException $e) {
+			if ($this->rootView->file_exists($target)) {
+				$this->rootView->unlink($target);
+			}
+			return false;
+		}
+
+		return true;
+	}
+
+	protected function decryptAllUsersFiles(ProgressBar $progress) {
+		$userList = [];
+
+		foreach ($this->userManager->getBackends() as $backend) {
+			$limit = 500;
+			$offset = 0;
+			do {
+				$users = $backend->getUsers('', $limit, $offset);
+				foreach ($users as $user) {
+					$userList[] = $user;
+				}
+				$offset += $limit;
+			} while (count($users) >= $limit);
+		}
+
+		$userNo = 1;
+		foreach ($userList as $uid) {
+			$this->decryptUsersFiles($uid, $progress);
+			$progress->advance();
+			$userNo++;
+		}
+		return true;
+	}
+
+	protected function decryptUsersFiles($uid, ProgressBar $progress) {
+		$this->setupUserFS($uid);
+		$directories = [];
+		$directories[] = '/' . $uid . '/files';
+
+		while ($root = array_pop($directories)) {
+			$content = $this->rootView->getDirectoryContent($root);
+			foreach ($content as $file) {
+				// only decrypt files owned by the user
+				if($file->getStorage()->instanceOfStorage('OCA\Files_Sharing\SharedStorage')) {
+					continue;
+				}
+				$path = $root . '/' . $file['name'];
+				if ($this->rootView->is_dir($path)) {
+					$directories[] = $path;
+					continue;
+				} else {
+					try {
+						if ($file->isEncrypted() !== false) {
+							if ($this->decryptFile($path) === false) {
+								$progress->setMessage("decrypt files for user $uid: $path (already decrypted)");
+								$progress->advance();
+							}
+						}
+					} catch (\Exception $e) {
+						if (isset($this->failed[$uid])) {
+							$this->failed[$uid][] = $path;
+						} else {
+							$this->failed[$uid] = [$path];
+						}
+					}
+				}
+			}
+		}
+
+		if (empty($this->failed)) {
```

Comment:
show error in case of failure ? an admin might want to ctrl+c when seeing this in real-time